### PR TITLE
Use offcanvas nav for mobile

### PR DESCRIPTION
**Pull request by dennisreimann**
Bootstrap v5 introduced the offcanvas navigation component, which makes for a nicer UX than our current collapsing approach.

offcanvas

I also tried having the topbar present all the time, so that just the navigation menu slides in (underneath the header/topbar), but this doesn't work without duplicating markup and it also behaves weird visually when scaling. So I ended up with this version and I think it works quite nice. 

Feedback welcome!

### CODE REVIEW

**Comment by dstrukt** [comment]
Will check out the branch, test a bit, and give some feedback - definitely looking much better!

**Review by pavlenex — APPROVED** [review]
tACK, great improvement. 🚀

**Review by dstrukt — APPROVED** [review]
LGTM.

You've already seen my thoughts in the MM, and we can discuss a few things in tomorrow's design call, but this Is a big improvement from the existing menu.

**Comment by dennisreimann** [comment]
Prepared another variant, which I like better in terms of animation style (we can further improve this into the direction you proposed) and positioning of logo and nav.

offcanvas-2

**Comment by dstrukt** [comment]
Yeah I'm a bigger fan of this variant as well.

I went and dug up the code I wrote a while, I think this is everything, had some weird tricks and hacks I was doing for the switching background color these components would sit above ... but for the most part pretty simple to isolate the parts that are relevant and helpful for a polished look:

`
@media (max-width:768px){
    .mobile-nav{
        position:absolute;
        width:100%;
        z-index:99999
    }
    .mobile-fixed{
        position:fixed
    }
}

.nav-toggle{
    display:none
}
@media (max-width:991px){
    .nav-toggle{
        display:block;
        position:relative;
        right:0;
        z-index:999999;
        padding:20px 28px 20px 16px;
        cursor:pointer
    }
    .nav-toggle:focus{
        outline:0
    }
    .nav-toggle span,.nav-toggle span:after,.nav-toggle span:before{
        content:'';
        position:absolute;
        display:block;
        width:28px;
        height:2px;
        border-radius:1px;
        background:#fff;
        cursor:pointer
    }
    .nav-toggle.dark-toggle span,.nav-toggle.dark-toggle span:after,.nav-toggle.dark-toggle span:before{
        content:'';
        position:absolute;
        display:block;
        width:28px;
        height:2px;
        border-radius:1px;
        background:#1d232e;
        cursor:pointer
    }
    .nav-toggle.light-toggle span,.nav-toggle.light-toggle span:after,.nav-toggle.light-toggle span:before{
        content:'';
        position:absolute;
        display:block;
        width:28px;
        height:2px;
        border-radius:1px;
        background:#fff;
        cursor:pointer
    }
    .nav-toggle span:before{
        top:-8px
    }
    .nav-toggle span:after{
        bottom:-8px
    }
    .nav-toggle span,.nav-toggle span:after,.nav-toggle span:before{
        -webkit-transition:all .3s ease-in-out;
        transition:all .3s ease-in-out
    }
    .nav-toggle.active span{
        background-color:transparent
    }
    .nav-toggle.active span:after,.nav-toggle.active span:before{
        top:0
    }
    .nav-toggle.active span:before{
        -webkit-transform:rotate(45deg);
        -ms-transform:rotate(45deg);
        transform:rotate(45deg)
    }
    .nav-toggle.active span:after{
        top:8px;
        -webkit-transform:translateY(-8px) rotate(-45deg);
        -ms-transform:translateY(-8px) rotate(-45deg);
        transform:translateY(-8px) rotate(-45deg)
    }
}
.overlay{
    position:fixed;
    top:0;
    left:0;
    width:100%;
    height:100%;
    background:#121212;
    z-index:1
}
.overlay nav{
    margin-top:100px;
    position:relative;
    top:50%;
    height:60%;
    text-align:left
}
.overlay ul{
    position:relative;
    display:block;
    margin:0 auto;
    padding:0;
    list-style:none
}
.overlay ul li{
    display:block
}
.overlay ul li:last-child{
    margin-right:0
}
.overlay ul li i{
    display:block;
    margin-bottom:10px;
    font-size:40px
}
.overlay ul li a{
    font-family:favoritregular;
    text-align:left;
    display:block;
    padding:12px 4px;
    color:#fff;
    text-transform:normal;
    font-weight:400;
    font-size:24px;
    -webkit-transition:all .3s ease-in-out;
    transition:all .3s ease-in-out
}
.overlay ul li a:focus,.overlay ul li a:hover{
    color:#ffbd00
}
.overlay ul li.active a{
    color:#ffbd00
}
.overlay-boxify{
    visibility:hidden;
    opacity:0;
    -webkit-transition:opacity .5s,visibility 0s .5s;
    transition:opacity .5s,visibility 0s .5s
}
.overlay-boxify.open{
    visibility:visible;
    opacity:1;
    -webkit-transition:opacity .5s;
    transition:opacity .5s
}
.overlay-boxify nav ul li{
    opacity:0;
    -webkit-transform:translate3d(0,10px,0);
    transform:translate3d(0,10px,0);
    -webkit-transition:-webkit-transform .5s,opacity .5s;
    transition:transform .5s,opacity .5s
}
.overlay-boxify.open nav ul:first-child li:first-child{
    -webkit-transition-delay:50ms;
    transition-delay:50ms
}
.overlay-boxify.open nav ul:first-child li:nth-child(2){
    -webkit-transition-delay:.1s;
    transition-delay:.1s
}
.overlay-boxify.open nav ul:first-child li:nth-child(3){
    -webkit-transition-delay:.15s;
    transition-delay:.15s
}
.overlay-boxify.open nav ul:first-child li:nth-child(4){
    -webkit-transition-delay:.2s;
    transition-delay:.2s
}
.overlay-boxify.open nav ul:first-child li:nth-child(5){
    -webkit-transition-delay:.25s;
    transition-delay:.25s
}
.overlay-boxify.open nav ul:first-child li:nth-child(6){
    -webkit-transition-delay:.3s;
    transition-delay:.3s
}
.overlay-boxify.open nav ul:first-child li:nth-child(7){
    -webkit-transition-delay:.3s;
    transition-delay:.3s
}
.overlay-boxify.open nav ul:first-child li:nth-child(8){
    -webkit-transition-delay:.3s;
    transition-delay:.3s
}
.overlay-boxify.open nav ul:first-child li:nth-child(9){
    -webkit-transition-delay:.3s;
    transition-delay:.3s
}
.overlay-boxify.open nav ul:first-child li:nth-child(10){
    -webkit-transition-delay:.3s;
    transition-delay:.3s
}
.overlay-boxify.open nav ul:first-child li:nth-child(11){
    -webkit-transition-delay:.3s;
    transition-delay:.3s
}
.overlay-boxify.open nav ul:first-child li:nth-child(12){
    -webkit-transition-delay:.3s;
    transition-delay:.3s
}
.overlay-boxify.open nav ul li{
    opacity:1;
    -webkit-transform:translate3d(0,0,0);
    transform:translate3d(0,0,0)
}

`

`
<div class="mobile-nav mobile-fixed" data-controller="navbar">
   
   <nav class="navbar navbar-expand-lg bg-transparent">
      <div class="container-fluid">

         <a class="navbar-brand" href="/">
            <SVG LOGO>
         </a>
         <a id="trigger-overlay" class="nav-toggle active" href="#">
	         <span></span>
         </a>

         <div class="collapse navbar-collapse" id="navbarSupportedContent">
            <ul class="navbar-nav ml-auto">
               <li class="nav-item ">
                  <a class="nav-link nav-link-light" href="">About</a>
               </li>
               <li class="nav-item ">
                  <a class="nav-link nav-link-light" href="">Compositions</a>
               </li>
               <li class="nav-item ">
                  <a class="nav-link nav-link-light" href="">Singing</a>
               </li>
               <li class="nav-item ">
                  <a class="nav-link nav-link-light" href="">Gallery</a>
               </li>
               <li class="nav-item ">
                  <a class="nav-link nav-link-light" href="">Concerts</a>
               </li>
               <li class="nav-item ">
                  <a class="nav-link nav-link-light" href="">Contact</a>
               </li>
            </ul>
         </div>
         
      </div>
   </nav>
   
   <div class="overlay overlay-boxify open">
      <div class="container">
         <nav>
            <ul>
               <li class="">
                  <a href="">About</a>
               </li>
               <li class="">
                  <a href="">Compositions</a>
               </li>
               <li class="">
                  <a href="">Singing</a>
               </li>
               <li class="">
                  <a href="">Gallery</a>
               </li>
               <li class="">
                  <a href="">Concerts</a>
               </li>
               <li class="">
                  <a href="">Contact</a>
               </li>
            </ul>
         </nav>
      </div>
   </div>
   
</div>

`

**Comment by dstrukt** [comment]
And for others, context on the direction, and reference for the code above's result:
Menu

**Comment by dennisreimann** [comment]
@dstrukt I tried animating the individual items -- it gets tricky for our case, because these are actually two separate lists/containers: The main nav items (Stores, Apps, etc.) and the meta items (Account, Notifications, Theme, Logout). We cannot address them across the lists (e.g. by using `:nth-of-type`) and we cannot address them by number. The former is reset per list/container, the latter has edge cases like user vs. admin (Server Settings being shown) and potentially plugins (e.g. LNbank will be in the main nav too).

TL;DR Let's keep it like this for now to avoid additional complexity. I wouldn't want to introduce JavaScript that numbers the items just for the animation effect.

**Comment by dstrukt** [comment]
Ahh interesting to note.

Sounds good, just sharing some of the things I did, and no worries, figured we'd pick and choose what was feasible.

This is already a huge improvement.

----

Not to complicate or even request, but was curious, WRT the first iteration, I could see that being an equally suitable solution if it displaced/pushed the main container (as opposed to the move in overlay), think twitter .. is that possible with our current setup / easy enough to do?

**Comment by dennisreimann** [comment]
>  is that possible with our current setup / easy enough to do?

That'd be possible, but it would require custom CSS and also JavaScript, because right now only the offcanvas classes are set – we'd need to set classes on a container that contains both the nav and the main content. So this would have to be managed on `body` level.

**Comment by NicolasDorier** [comment]
is it ready to be merged?

**Comment by pavlenex** [comment]
Please do not go with crazy animations here ,we don't need anything fancy. The first itteration is good enough imo, all after it are too much.

**Comment by dstrukt** [comment]
+1 LGTM, but more eyes the better.

Great work! @dennisreimann
